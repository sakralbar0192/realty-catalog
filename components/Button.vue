<template>
  <button
    :type="props.type"
    :class="buttonClasses"
    :disabled="props.disabled"
    :aria-label="props.ariaLabel"
    @click="$emit('click')"
  >
    <slot></slot>
  </button>
</template>

<!-- Styles should be in separate SCSS files, not in <style> blocks -->
<script setup lang="ts">
import { computed } from 'vue'
import styles from '~/assets/styles/components/Button.module.scss'

interface Props {
  size?: 'small' | 'medium' | 'large'
  variant?: 'primary' | 'secondary'
  type?: 'button' | 'submit' | 'reset'
  disabled?: boolean
  ariaLabel?: string
}

const props = withDefaults(defineProps<Props>(), {
  size: 'medium',
  variant: 'primary',
  type: 'button'
})

defineEmits<{
  click: []
}>()

const buttonClasses = computed(() => [
  styles.button,
  styles[`button--${props.variant}`],
  styles[`button--${props.size}`]
])
</script>
