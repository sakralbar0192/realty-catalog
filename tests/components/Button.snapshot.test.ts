import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import Button from '~/components/Button.vue'

describe('Button snapshots', () => {
  it('matches default snapshot', () => {
    const wrapper = mount(Button, {
      slots: {
        default: 'Default Button',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('matches secondary variant snapshot', () => {
    const wrapper = mount(Button, {
      props: {
        variant: 'secondary',
      },
      slots: {
        default: 'Secondary Button',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('matches disabled state snapshot', () => {
    const wrapper = mount(Button, {
      props: {
        disabled: true,
      },
      slots: {
        default: 'Disabled Button',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
