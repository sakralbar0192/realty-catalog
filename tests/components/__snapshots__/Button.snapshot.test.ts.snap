// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button snapshots > matches default snapshot 1`] = `"<button type="button" class="_button_498b9f _button--primary_498b9f _button--medium_498b9f">Default Button</button>"`;

exports[`Button snapshots > matches disabled state snapshot 1`] = `"<button type="button" class="_button_498b9f _button--primary_498b9f _button--medium_498b9f" disabled="">Disabled Button</button>"`;

exports[`Button snapshots > matches secondary variant snapshot 1`] = `"<button type="button" class="_button_498b9f _button--secondary_498b9f _button--medium_498b9f">Secondary Button</button>"`;
